src/libasr/pass/print_list.cpp

```diff
@@ -52,148 +52,186 @@ class PrintListVisitor
-
-            std::string list_iter_var_name;
-            ASR::symbol_t *list_iter_variable;
-            ASR::expr_t *list_iter_var;
-            {
-                list_iter_var_name =
-                    current_scope->get_unique_name("__list_iterator");
-                list_iter_variable =
-                    ASR::down_cast<ASR::symbol_t>(ASR::make_Variable_t(
-                        al, x.base.base.loc, current_scope,
-                        s2c(al, list_iter_var_name), nullptr, 0, ASR::intentType::Local,
-                        nullptr, nullptr, ASR::storage_typeType::Default,
-                        int_type, ASR::abiType::Source, ASR::accessType::Public,
-                        ASR::presenceType::Required, false));
-
-                current_scope->add_symbol(list_iter_var_name,
-                                          list_iter_variable);
-                list_iter_var = ASRUtils::EXPR(
-                    ASR::make_Var_t(al, x.base.base.loc, list_iter_variable));
-            }
+    void print_list_helper(ASR::expr_t *list_expr, ASR::expr_t *sep_expr,
+                ASR::expr_t *end_expr, const Location &loc) {
+        ASR::List_t *listC =
+                ASR::down_cast<ASR::List_t>(ASRUtils::expr_type(list_expr));
+        ASR::ttype_t *int_type = ASRUtils::TYPE(
+                ASR::make_Integer_t(al, loc, 4, nullptr, 0));
+        ASR::ttype_t *bool_type = ASRUtils::TYPE(
+            ASR::make_Logical_t(al, loc, 4, nullptr, 0));
+        ASR::ttype_t *str_type_len_0 = ASRUtils::TYPE(ASR::make_Character_t(
+            al, loc, 1, 0, nullptr, nullptr, 0));
+        ASR::ttype_t *str_type_len_1 = ASRUtils::TYPE(ASR::make_Character_t(
+            al, loc, 1, 1, nullptr, nullptr, 0));
+        ASR::ttype_t *str_type_len_2 = ASRUtils::TYPE(ASR::make_Character_t(
+            al, loc, 1, 2, nullptr, nullptr, 0));
+        ASR::expr_t *comma_space =
+            ASRUtils::EXPR(ASR::make_StringConstant_t(
+                al, loc, s2c(al, ", "), str_type_len_2));
+        ASR::expr_t *single_quote =
+            ASRUtils::EXPR(ASR::make_StringConstant_t(
+                al, loc, s2c(al, "'"), str_type_len_1));
+        ASR::expr_t *empty_str = ASRUtils::EXPR(ASR::make_StringConstant_t(
+            al, loc, s2c(al, ""), str_type_len_0));
+        ASR::expr_t *open_bracket =
+            ASRUtils::EXPR(ASR::make_StringConstant_t(
+                al, loc, s2c(al, "["), str_type_len_1));
+        ASR::expr_t *close_bracket =
+            ASRUtils::EXPR(ASR::make_StringConstant_t(
+                al, loc, s2c(al, "]"), str_type_len_1));
+
+        std::string list_iter_var_name;
+        ASR::symbol_t *list_iter_variable;
+        ASR::expr_t *list_iter_var;
+        {
+            list_iter_var_name =
+                current_scope->get_unique_name("__list_iterator");
+            list_iter_variable =
+                ASR::down_cast<ASR::symbol_t>(ASR::make_Variable_t(
```

Comment:
Use the following function, 

https://github.com/lcompilers/lpython/blob/ed807bbf11ec00e42a4a144adef48e6c67589ffe/src/libasr/pass/pass_utils.h#L53-L54